libraries/stdlib/jdk8/src/kotlin/random/jdk8/PlatformThreadLocalRandom.kt

```diff
@@ -8,15 +8,20 @@ package kotlin.random.jdk8
 import java.util.concurrent.ThreadLocalRandom
 
 @Suppress("INVISIBLE_REFERENCE", "INVISIBLE_MEMBER", "CANNOT_OVERRIDE_INVISIBLE_MEMBER")
-internal class PlatformThreadLocalRandom : kotlin.random.AbstractPlatformRandom() {
+internal object PlatformThreadLocalRandom : kotlin.random.AbstractPlatformRandom(), Serializable {
+    private const val serialVersionUID = -8182665159126504406L
+
     // TODO no bridge generated for covariant override
-    override val impl: java.util.Random get() = ThreadLocalRandom.current()
+    override val impl: java.util.Random
+        get() = ThreadLocalRandom.current()
 
     override fun nextInt(from: Int, until: Int): Int = ThreadLocalRandom.current().nextInt(from, until)
     override fun nextLong(until: Long): Long = ThreadLocalRandom.current().nextLong(until)
     override fun nextLong(from: Long, until: Long): Long = ThreadLocalRandom.current().nextLong(from, until)
     override fun nextDouble(until: Double): Double = ThreadLocalRandom.current().nextDouble(until)
 
-//     do not delegate this, as it's buggy in JDK8+ (up to 11 at the moment of writing)
-//     override fun nextDouble(from: Double, until: Double): Double = ThreadLocalRandom.current().nextDouble(from, until)
-}
\ No newline at end of file
+    //     do not delegate this, as it's buggy in JDK8+ (up to 11 at the moment of writing)
+    //     override fun nextDouble(from: Double, until: Double): Double = ThreadLocalRandom.current().nextDouble(from, until)
+
+    private fun readResolve(): Any = PlatformThreadLocalRandom
```

Comment:
I think such internal implementations should not be serializable at all. 
`Random.Default` manages its singleton serialization entirely and therefore it shouldn't expose any implementation details in a serialized form.